src/components/Carousel/ControllableCarousel.tsx

```diff
@@ -0,0 +1,96 @@
+'use client';
+
+import { usePrevNextButtons } from '@/hooks/usePrevNextButtons';
+import useEmblaCarousel from 'embla-carousel-react';
+import { IndicatorButton, useIndicatorButton } from './CarouselIndicator';
+
+import { ReactNode } from 'react';
+import ChevronLeft from '../Icon/icons/ChevronLeft';
+import ChevronRight from '../Icon/icons/ChevronRight';
+
+type ControlledCarouselPropsType<T> = {
+  slides: T[];
+  renderSlide: (card: T, index: number) => ReactNode;
+  spaceSize?: 's' | 'm' | 'l';
+};
+
+const ControlledCarousel = <T,>({
+  slides,
+  renderSlide,
+  spaceSize = 'm',
+}: ControlledCarouselPropsType<T>) => {
+  const [emblaRef, emblaApi] = useEmblaCarousel({
+    dragFree: true,
+  });
+
+  const { selectedIndex, scrollSnaps, onIndicatorButtonClick } =
+    useIndicatorButton(emblaApi);
+
+  const {
+    prevBtnDisabled,
+    nextBtnDisabled,
+    onPrevButtonClick,
+    onNextButtonClick,
+  } = usePrevNextButtons(emblaApi);
+
+  const [gapSizeClassName, padddingSizeClassName] =
+    spaceSize === 'l'
+      ? ['ml-[-45px] mobile:ml-[-50px]', 'pl-[45px] mobile:pl-[50px]']
+      : spaceSize === 'm'
+        ? ['ml-[-35px] mobile:ml-[-45px]', 'pl-[35px] mobile:pl-[45px]']
+        : ['ml-[-30px]', 'pl-[30px]'];
+
+  return (
+    <section className='relative flex flex-col gap-[90px]'>
+      <div className='absolute -top-[120px] right-0  hidden tablet:flex'>
+        <button
+          type='button'
+          className='touch-manipulation  cursor-pointer w-[60px] h-[60px] rounded-full  disabled:text-gray-800'
+          onClick={onPrevButtonClick}
+          disabled={prevBtnDisabled}
+        >
+          <ChevronLeft
+            className={`${prevBtnDisabled ? 'text-gray-800' : 'text-gray-200'}`}
+          />
+        </button>
+        <button
+          type='button'
+          className='touch-manipulation  cursor-pointer w-[60px] h-[60px] rounded-full  disabled:text-gray-800'
+          onClick={onNextButtonClick}
+          disabled={nextBtnDisabled}
+        >
+          <ChevronRight
+            className={`${nextBtnDisabled ? 'text-gray-800' : 'text-gray-200'}`}
```

Comment:
```suggestion
            <Icon name= 'ChevronRight' size='s' className={`${nextBtnDisabled ? 'text-gray-800' : 'text-gray-200'} />
```

`ChevronLeft`, `ChevronRight` 모두 `Icon` 컴포넌트로 작성하지 않으신 이유가 있나요? 위 코드는 `Icon` 컴포넌트로 작성했을 경우의 예시 코드입니다 :)

Reply:
이 부분은 각 Icon을 직접 호출해서 사용하는 것으로 생각하고 잘못 사용하고 있었습니다;; 이부분 수정하겠습니다!

Reply:
ㅋㅋㅋㅋㅋㅋㅋ 넵넵 천천히 반영해주셔용